modules/kmm-replacing-in-tree-modules-with-out-of-tree-modules.adoc

```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// * hardware_enablement/kmm-kernel-module-management.adoc
+
+:_content-type: CONCEPT
+[id="kmm-replacing-in-tree-modules-with-out-of-tree-modules_{context}"]
+= Replacing in-tree modules with out-of-tree modules
+
+You can use Kernel Module Management (KMM) to build kernel modules that can be loaded or unloaded into the kernel on demand. These modules extend the functionality of the kernel without the need to reboot the system.
+Modules can be configured as built-in or dynamically loaded.
+
+Dynamically loaded modules include in-tree modules and out-of-tree (OOT) modules.
+In-tree modules are internal to the Linux kernel tree, that is, they are already part of the kernel.
+Out-of-tree modules are external to the Linux kernel tree. They are generally written for
+development and testing purposes, such as testing the new version of a kernel module that is
+shipped in-tree, or to deal with incompatibilities.
+
+Some modules loaded by KMM could replace in-tree modules already loaded on the node.
+To unload an in-tree module before loading your module,
+set the `.spec.moduleLoader.container.inTreeModuleToRemove`. This is an example for module replacement
+for all kernel mappings:
+
+[source,yaml]
+----
+spec:
+  moduleLoader:
+    container:
+      modprobe:
```

Comment:
@qbarrand Do we need to show usage of `modprobe -r <module_name>` and `modprobe -d <module_name>`?  Any other changes to these examples?

Reply:
I believe we do not need to include that.

Reply:
Agree with @qbarrand on this one, This should not be included in the docs. 
If a customer really wants it, he can look at the code.

Reply:
I think the YAML example is fine. I see no reason to reference or explain the `modprobe` usage in this section.

Reply:
@qbarrand To be clear should I also remove `moduleName: mod_a`? 
![image](https://github.com/openshift/openshift-docs/assets/5238708/2decac3b-b02a-41ea-b64f-0d63eb4f3412)